lego/apps/email/notifications.py

```diff
@@ -0,0 +1,101 @@
+from datetime import timedelta
+
+from django.utils import timezone
+
+from lego.apps.events.constants import EVENT_TYPE_TRANSLATIONS
+from lego.apps.events.models import Event
+from lego.apps.joblistings.constants import JOB_TYPE_TRANSLATIONS
+from lego.apps.joblistings.models import Joblisting
+from lego.apps.notifications.constants import WEEKLY_MAIL
+from lego.apps.notifications.notification import Notification
+from lego.apps.permissions.constants import VIEW
+from lego.apps.tags.models import Tag
+
+
+class WeeklyNotification(Notification):
+
+    name = WEEKLY_MAIL
+
+    def generate_mail(self):
+        yesterday_timestamp = timezone.now() - timedelta(days=1)
+        last_sunday_timestamp = timezone.now() - timedelta(days=7)
+
+        weekly_tag = Tag.objects.filter(tag="weekly").first()
+        # Check if weekly tag exists so it does not crash if some idiot deletes the weekly tag
+        todays_weekly = (
+            weekly_tag.article_set.filter(created_at__gt=yesterday_timestamp).first()
+            if weekly_tag
+            else None
+        )
+
+        week_number = timezone.now().isocalendar().week
+
+        events_next_week = Event.objects.filter(
+            pools__activation_date__gt=timezone.now(),
+            pools__activation_date__lt=timezone.now() + timedelta(days=7),
+        ).distinct()
+
+        filtered_events = filter(
+            lambda event: self.user.has_perm(VIEW, obj=event), events_next_week
+        )
```

Comment:
You _could_ also use the `filter_queryset` method on the permission handler, but this shouldn't be too big (and runs async) so it's fine I guess.